v2/concordium/types.proto

```diff
@@ -503,6 +503,58 @@ message Cooldown {
   CooldownStatus status = 3;
 }
 
+// The unique id of a token. The identifier is the position in the token table.
+message TokenId {
+  uint64 value = 1;
+}
+
+// The unique identifier of a token module.
+message TokenModuleId {
+  uint64 value = 1;
+}
+
+// PLT amount representation.
+message TokenAmount {
+  uint64 amount = 1;
+  // Number of decimals in the representation.
+  uint8 decimal = 2;
+}
+
+// Protocol level token (PLT).
+message Token {
+  TokenId token_id = 1;
+  TokenAmount balance = 2;
+}
+
+message TokenInfo {
+  // The unique token id.
+  string token_id = 1;
+  // The unique token symbol.
+  string token_symbol = 2;
+  // The token name. This should be unique, but uniqueness is enforced only by
+  // the chain governance, not by the consensus protocol.
+  string token_name = 3;
+  // The id of the module implementing this token.
+  TokenModuleId token_module_id = 4;
+  // Number of decimals in the token amount representation.
+  uint8 decimal = 5;
+  // A URL pointing to additional meta information about the token.
+  optional string token_meta_url = 6;
+  // Account address of the issuer. The issuer is the holder of the nominated
+  // account which can perform token-governance operations.
+  AccountAddress issuer = 7;
+  // Whether the token supports an allow list.
+  bool supports_allow_list = 8;
+  // Whether the token supports a dely list.
+  bool supports_deniy_list = 9
+  // Whether the token supports minting.
+  bool supports_minting = 10;
+  // Whether the token supports burning.
+  bool supoorts_burning = 11;
```

Comment:
Not sure how these `supports_*` will scale when token modules become wasm and new features are introduced.

Reply:
Agreed, if we want scalability a bitmap would be better. But I'm not sure if we should design with this in mind for the MVP already.

Reply:
I implemented this now as a sequence of feature flags.